packages/smooth_app/lib/pages/folksonomy/folksonomy_provider.dart

```diff
@@ -57,9 +63,19 @@ class FolksonomyProvider extends ValueNotifier<FolksonomyState> {
     }
   }
 
+  // Display tags from local database first (to see it offline), then update from API.
   Future<void> fetchProductTags() async {
     try {
-      value = const FolksonomyStateLoading();
+      final List<ProductTag> localTags = await DaoFolksonomy(
+        localDatabase,
+      ).getTags(barcode);
```

Comment:
```suggestion
      final DaoFolksonomy daoFolksonomy = DaoFolksonomy(localDatabase);
      final List<ProductTag> localTags = await daoFolksonomy.get(barcode);
```
We're reusing `daoFolksonomy` later.